### PR TITLE
Bridge and Anchor-Handler Pallet Implementation

### PR DESCRIPTION
**Pull request by dev0x1**
We have added and modified chainbridge governance to darkwebb-solidity and want to do the same to darkwebb-substrate. The solidity repo can be found at [https://github.com/webb-tools/darkwebb-solidity/blob/main/contracts/handlers/AnchorHandler.sol](https://github.com/webb-tools/darkwebb-solidity/blob/main/contracts/handlers/AnchorHandler.sol).

### CODE REVIEW

**Comment by drewstone** [comment]
Can we name the directory `bridge`?

File: pallets/chainbridge/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+[dependencies.codec]
+default-features = false
+features = ["derive"]
+package = "parity-scale-codec"
+version = "2.0.0"
+
+[dependencies]
+# third-party dependencies
+serde = { version = "1.0.101", optional = true }
+chainbridge = { version = "0.0.2", git = "https://github.com/ChainSafe/chainbridge-substrate.git", rev = "3999027", default-features = false}
```
Comment (drewstone) [inline pallets/chainbridge/Cargo.toml:24]: we don't want to depend on their pallet at all. I suggest copying it entirely over into its own pallet. We will need to modify it enough that this makes sense.

**Review by drewstone — CHANGES_REQUESTED** [review]
We will want to make changes to the OG chainbridge design enough that we should clone the base pallet into our repo as well @dev0x1 .

File: pallets/chainbridge/Cargo.toml (hunk 1 above)
Comment (dev0x1) [reply pallets/chainbridge/Cargo.toml]: Yeah, I also feel so. I will copy it in a `chainbridge` pallet then.

File: pallets/anchor-handler/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+[build-dependencies]
+wasm-builder-runner = { version = "2.0.0", package = "substrate-wasm-builder-runner" }
+[features]
+default = ["std"]
+std = [
+	"codec/std",
+	"serde",
+	"sp-std/std",
+	"sp-runtime/std",
+    "sp-io/std",
```
Comment (drewstone) [inline pallets/anchor-handler/Cargo.toml:59]: spacing is off

File: pallets/anchor-handler/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(_n: BlockNumberFor<T>) {
+			// FIXME create/init anchors
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(195_000_000)]
+		pub fn set_maintainer(origin: OriginFor<T>, new_maintainer: T::AccountId) -> DispatchResultWithPostInfo {
```
Comment (drewstone) [inline pallets/anchor-handler/src/lib.rs:142]: Is maintainer being used for anything here? usually this was for pallets with parameters. If not we can remove.

**Review by drewstone — CHANGES_REQUESTED** [review]
Did you merge master? Seems there's a lot of files that should already be in repo being committed again here.

**Pull request merged**